### DISCUSSION TITLE
CentOS, STATIC_ROOT, and SELinux (Documentation update request)

### DISCUSSION THREAD

**Discussion by Astlor** [netbox-community/netbox#6924] [General]
I don't know what the appropriate form, or forum, or whatever is required for this*, but you are missing a critical piece of "how to make the thing work" on https://netbox.readthedocs.io/en/stable/installation/5-http-server/

If you have:

1) CentOS 8 (or related)
2) Apache

Then you need to run 

chcon -R -t httpd_sys_content_t /opt/netbox/netbox/static/

or SELinux will give you the dreaded http://netbox.must.lab/media-failure/?filename=css/base.css error.

*I note, based on attempting to search for the answer to "what is this error and why am I getting it" that multiple "top" responses on Google were someone asking a very clearly worded "why is this not working" question, complete with details, and then getting told by one of the community members "this won't be answered because it isn't in the right format/submitted via the right forum" etc.  So I'm guessing that providing the answer here will probably get sucked into the same rage-inducing [expletive deleted] void, but I refuse to be DenverCoder9 (for reference: https://xkcd.com/979/).

So the above is the answer to at least one variation of "The HTTP service (e.g. nginx or Apache) is configured to serve files from the STATIC_ROOT path", for CentOS 8 using Apache.  (And yes, you still need to run this, even if you do the setsebool thing that's already in the documentation.)

Thank you, and have a good day.

**Comment by yrro** [comment]
As is, that change won't survive a relabel. You can make it permanent with:

```
# semanage fcontext -a -t httpd_sys_content_t "/opt/netbox/netbox/static(/.*)?"
```

You can use `matchpathcon` on a path within that directory before and after to see the effect.

For more information, see "Labeling problems" at <https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/using_selinux/troubleshooting-problems-related-to-selinux_using-selinux#fixing-an-analyzed-selinux-denial_troubleshooting-problems-related-to-selinux>.
